sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/TestCases/UsingRegularExpressions.cs

```diff
@@ -7,65 +7,81 @@ namespace Tests.Diagnostics
 {
     class Program
     {
-        void Main()
+        void Main(string s)
         {
             Regex r;
-            r = new Regex(""); // Noncompliant {{Make sure that using a regular expression is safe here.}}
-//              ^^^^^^^^^^^^^
-            r = new Regex("", RegexOptions.Compiled); // Noncompliant
-            r = new Regex("", RegexOptions.Compiled, TimeSpan.Zero); // Noncompliant
+            r = new Regex(""); // Compliant, less than 3 characters
+            r = new Regex("**"); // Compliant, less than 3 characters
+            r = new Regex("+*"); // Compliant, less than 3 characters
+            r = new Regex("abcdefghijklmnopqrst"); // Compliant, does not have the special characters
+            r = new Regex("abcdefghijklmnopqrst+"); // Compliant, has only 1 special character
+            r = new Regex("{abc}+defghijklmnopqrst"); // Noncompliant
+            r = new Regex("{abc}+{a}"); // Noncompliant {{Make sure that using a regular expression is safe here.}}
+//              ^^^^^^^^^^^^^^^^^^^^^^
+            r = new Regex("+++"); // Noncompliant
```

Comment:
I guess we'll raise false positives in the cases where special characters exist in the string but those characters have been escaped. 
That's probably ok for now (unless it's easy to fix), but it would be worth adding a a couple of test cases to document the FNs.